### DISCUSSION TITLE
请求无名大佬的帮助

### DISCUSSION THREAD

**Discussion by oncall12** [imsyy/home#124] [Q&A] [answered]
![微信截图_20230421143549](https://user-images.githubusercontent.com/131435794/233569994-edb001e2-e691-41ab-a0bb-12d8930f8e90.jpg)

请问怎样才能改成点击音乐直接打开网站原有的播发器音乐列表呢？我是小白一个，一个人瞎鼓捣了研究大半天了，硬是没搞懂，所以才来请求大佬的帮助，十万分感谢！！
`                                    <div class="col">
                                        <a id="link-url-3" href="https://music.imsyy.top/" target="_blank">
                                            <div class="link-card cards">
                                                <i id="link-icon-3" class="fa-solid fa-music"></i>
                                                <span class="link-name" id="link-name-3">音乐</span>
                                            </div>
                                        </a>
                                    </div>`

**Comment by imsyy** [comment]
啊，这可不好改 😂，音乐组件和网站列表不是在一起的，要改的话会涉及改动多个文件，我稍后把改动的文件内容发给你，你参考一下

**Reply by oncall12** [reply]
我的意思是点击音乐，然后它就打开自带的音乐列表，和正常点击音乐列表一样显示就行，不一定出现在图片这个位置

**Reply by imsyy** [reply]
> 我的意思是点击音乐，然后它就打开自带的音乐列表，和正常点击音乐列表一样显示就行，不一定出现在图片这个位置

我知道

**Reply by oncall12** [reply]
我还以为直接调用某个变量函数就行了呢，怪不得我今天一直坐在电脑旁研究都不行，╮(╯▽╰)╭

**Reply by imsyy** [reply]
> 我还以为直接调用某个变量函数就行了呢，怪不得我今天一直坐在电脑旁研究都不行，╮(╯▽╰)╭

已经把这个新功能推送到 `dev` 分支了，其他改动有点多 😂，这个功能具体是这要改动这两个文件：`src\components\Music\index.vue`，`src\components\Links\index.vue`，你试着比较一下

**Reply by oncall12** [reply]
我怎么没有看到文件呢 奇怪了，不知道在哪里下载

**Reply by imsyy** [reply]
> 我怎么没有看到文件呢 奇怪了，不知道在哪里下载

https://github.com/imsyy/home/tree/dev

**Reply by oncall12** [reply]
点进去没看懂 好尴尬啊 O(∩_∩)O

**Reply by imsyy** [reply]
> 点进去没看懂 好尴尬啊 O(∩_∩)O

🤣

**Reply by oncall12** [reply]
github上面都是英文，我都不会看，都不知道大佬帮我改的是哪个文件，也不知道下载哪个你修改过的，好烦

**Reply by imsyy** [reply]
> github上面都是英文，我都不会看，都不知道大佬帮我改的是哪个文件，也不知道下载哪个你修改过的，好烦

改了这两个文件：

- [Music/index.vue](https://github.com/imsyy/home/blob/dev/src/components/Music/index.vue)
- [Links/index.vue](https://github.com/imsyy/home/blob/dev/src/components/Links/index.vue)

**Reply by oncall12** [reply]
[3.1.3 版](https://github.com/imsyy/home/releases/tag/v3.1.3)是不是没有这两个文件

**Reply by imsyy** [reply]
有的，都在 `src/components` 目录下

**Reply by oncall12** [reply]
[3.1.3 版](https://github.com/imsyy/home/releases/tag/v3.1.3)没有这个目录

**Reply by imsyy** [reply]
> [3.1.3 版](https://github.com/imsyy/home/releases/tag/v3.1.3)没有这个目录

哦，原来的静态版本确实没有，你要改的是这个版本啊

**Reply by imsyy** [reply, marked as answer]
> [3.1.3 版](https://github.com/imsyy/home/releases/tag/v3.1.3)没有这个目录

把音乐 div 包括外面的 a 标签换成下面的代码

```html
<a onclick="document.getElementById('music-open').click()">
    <div class="link-card cards">
        <i id="link-icon-3" class="fa-solid fa-music"></i>
        <span class="link-name" id="link-name-3">音乐</span>
    </div>
</a>
```

**Reply by oncall12** [reply]
我研究研究，感谢大佬

**Reply by oncall12** [reply]
可以了，不过有个问题，手机端它不会弹出

**Reply by haoge0748** [reply]
手机端点击音乐弹出内置播放器的问题解决了吗
